packages/api/src/workers/check-video-streams.ts

```diff
@@ -0,0 +1,218 @@
+  }
+
+  return '';
+};
+
+const checkVideoOptions = (youTubeVideoItems: YouTubeVideoItem[]) =>
+  youTubeVideoItems.reduce<Dictionary<string>>((mapping, item) => {
+    return {
+      ...mapping,
+      [item.id]: getError(item),
+    };
+  }, {});
+
+const sendSlackMessage = (payload: SlackMessage, token: string) => {
+  return axios({
+    url: 'https://slack.com/api/chat.postMessage',
+    method: 'post',
+    data: payload,
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
+};
+
+export const checkVideoStreams = async (
+  connection: Connection,
+  projectId: string,
+) => {
+  const apiKey = process.env.FIREBASE_KEY;
+  const slackToken = process.env.SLACK_BOT_TOKEN;
+  const slackChannel = process.env.SLACK_BOT_CHANNEL;
+
+  // Check that the all necessary environment variables are set
+  if (!apiKey) {
+    logger.error('No google api key was defined');
+    return;
+  }
+
+  if (!slackToken) {
+    logger.error('No slack bot token was defined');
+    return;
+  }
+
+  if (!slackChannel) {
+    logger.error('No slack target channel was defined');
+    return;
+  }
+
+  // Fetch reefs with streams
+  const reefsWithStream = await connection.getRepository(Reef).find({
+    where: { videoStream: Not(IsNull()) },
+  });
+
+  // Extract the youTube id from the URLs
+  const youTubeIdToVideoStreamDetails = reefsWithStream.reduce<
+    reefIdToYouTubeIdMap
+  >((mapping, reef) => {
+    const id = getYouTubeVideoId(reef.videoStream!);
+
+    if (!id) {
+      logger.error(
+        `No valid YouTube id could be extracted for reef with id ${reef.id} and video stream ${reef.videoStream}`,
+      );
+      return mapping;
+    }
+
+    return {
+      ...mapping,
+      [id]: {
+        reefId: reef.id,
+        url: reef.videoStream!,
+      },
+    };
+  }, {});
+
+  const youTubeIds = Object.keys(youTubeIdToVideoStreamDetails);
+
+  // Fetch the youTube video information for each id
+  const axiosResponse = await fetchVideoDetails(youTubeIds, apiKey);
+
+  // Validate that the streams are valid
+  // For ids with no errors an empty string is returned
+  const youTubeIdToError = checkVideoOptions(axiosResponse.data.items);
+
+  // Create a simple alert template for slack
+  const messageTemplate = {
+    // The channel id is fetched by requesting the list on GET https://slack.com/api/conversations.list
+    // (the auth token should be included in the auth headers)
+    channel: slackChannel,
+    blocks: [
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: `Scheduled check of video streams in *${projectId}* instance`,
+        },
+      },
+      {
+        type: 'divider',
+      },
+      ...youTubeIds
+        .map((id) => {
+          const { reefId, url } = youTubeIdToVideoStreamDetails[id];
+
+          if (id in youTubeIdToError && !youTubeIdToError[id]) {
+            return undefined;
+          }
+
+          const template = {
```

Comment:
Why make a const if its not used? Consider just using a return
```suggestion
          return {
```

Reply:
I changed the iteration from a map to reduce to avoid the filtering. So now the const is needed, in my opinion, because it feels cleaner